src/gui/ExportProjectDialog.cpp

```diff
@@ -194,9 +194,13 @@ ProjectRenderer::ExportFileFormats convertIndexToExportFileFormat(int index)
 	case 0:
 		return ProjectRenderer::WaveFile;
 	case 1:
+#ifdef LMMS_HAVE_OGGVORBIS
 		return ProjectRenderer::OggFile;
 	case 2:
+#endif
+#ifdef LMMS_HAVE_MP3LAME
 		return ProjectRenderer::MP3File;
+#endif // If neither ogg nor mp3 are available, index 1 is OOB; fall through to default case
```

Comment:
Agreed. But I don't think parsing strings is the best solution either. I think it's best to use the User Data field in QItem to tag each combo box item directly with the ProjectRenderer::ExportFileFormat, converted to int. That way, we can extract this value in onFileFormatChanged, convert it back to the enum type, and work directly with it. This would also make this dialog a lot more extensible for the purpose of adding more renderers.  
I'll work on it and update this pull request once I have something that works.